### DISCUSSION TITLE
Rows in dashboards can be hard to use

### DISCUSSION THREAD

**Discussion by joshhunt** [grafana/grafana#42308] [Dashboards, Panels & Visualizations]
Rows are a really useful tool in Dashboards create rhythm and organisation, and to separate panels, However, sometimes they are bit fiddly and difficult to use. Some problems I've ran into myself:

 - Rows can only be reordered while they're collapsed. This makes it difficult to add a new row to the bottom of your Dashboard.
 - [Rows are "hungry"](https://github.com/grafana/grafana/issues/22368), where collapsing a row will hide all panels after it until the next row. This can create awkwardness especially when adding new rows to a dashboard which has un-rowed panels at the top.
 - And a few awkward interactions that I've ran into but can't remember 😅 

If you've had some problems with them, or have any other feedback with rows, please leave it below!

**Comment by joshhunt** [comment]
Just like how you can create a snapshot of a dashboard or a panel, perhaps it would be neat to snapshot or even link to just a group panels within a dashboard.

**Comment by ajenez** [comment]
Hi @joshhunt I've been handling this recently and I'm not sure what the best way to make a dashboard containing panels and rows is. I thought the "hungry" rows would only be an issue in the creation portion of the tool. However, it seems that even when I've designed the dashboard to have a collapsed row with a few panels and a single panel underneath that and have added it to my site, if I ever open the row the single panel will be swallowed. Not sure if there's a work around.

**Comment by the-it** [comment]
I had a similar experience as user @joshhunt when trying to duplicate a panel row. I initially tried to duplicate the row, but it wasn't feasible, so I opted to add a new empty row. However, I encountered a problem when some panels at the top of the dashboard were not in a row. It took me a while to figure out that the newly added row incorporated those un-rowed panels. I had to manually drag them out and move the row to the bottom of the dashboard. Overall, the process was not very intuitive. I would appreciate an entry to the row context menu "Adding row above/below".

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
